modules/ml/src/ann_mlp.cpp

```diff
@@ -946,7 +946,7 @@ class ANN_MLPImpl CV_FINAL : public ANN_MLP
             if( idx == 0 )
             {
                 //printf("%d. E = %g\n", iter/count, E);
-                if( fabs(prev_E - E) < epsilon )
+                if( fabs(prev_E - E) < epsilon && termCrit.type & CV_TERMCRIT_EPS )
```

Comment:
I believe it is better set epsilon to zero during initialization on line 902:
```
double epsilon = (termCrit.type & TermCriteria::EPS)
        ? termCrit.epsilon*count
        : 0;  // TermCriteria::COUNT
```

Also it is better to avoid deprecated C-style defines.

Reply:
I've changed the code accordingly, please check.